pkg/cloud/metadata_k8s.go

```diff
@@ -32,6 +49,21 @@ var DefaultKubernetesAPIClient = func() (kubernetes.Interface, error) {
 	return clientset, nil
 }
 
+// Get testing kubernetes APIclient
+var TestingKubernetesAPIClient = func() (kubernetes.Interface, error) {
+	// Get configuration from config file
+	config, err := clientcmd.BuildConfigFromFlags("", "/root/.kube/config")
```

Comment:
Agree with you.
@mkumatag what is the approach you suggest?
Should we pass the path for kubeconfig file from the driver deployment file?
(like how we use to pass Pam instance id earlier).
Or can we label the node with kubeconfig file path?

Reply:
But sometime in future we may need to run tests from outside the cluster.

Reply:
how other providers are handling it?

Reply:
This use case(TestingKubernetesAPIClient) is specific to our PowerVS cloud.

Inorder to create cloud object we need cloud instance id
```
cloud, err := powervscloud.NewPowerVSCloud(metadata.GetCloudInstanceId(), true)
```

The cloud object is used in test cases.
Another expectation I confirmed now is the test suite is going to be run inside the cluster.

As we need to get cloudinstanceId, I had decided to collect cloudInstanceId from metadata library.
Thus I added TestingKubernetesClient as part of metadata library.

I found a standard way for getting config.
Which is as below:
```
var (
       kubeConfig = os.Getenv("HOME") + ".kube/config"
)
config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
```

Reply:
I prefer having logic for covering all the logic running in-cluster or kubeconfig

Reply:
This function is very helpful in such situation, see is this works for this case - 

https://github.com/kubernetes-sigs/controller-runtime/blob/8f43f26d53ffb08a96ffce1298df0cd761f8be0d/alias.go#L78

Reply:
Ok. I will add logic for loading config from $HOME/.kube/config file if any error in loading in cluster config in DefaultKubernetesAPIClient itself and remove TestingKubernetesAPIClient.

Reply:
> 

Will check this

Reply:
The below code can fix the issue.

```
        // creates the in-cluster config
        config, err := rest.InClusterConfig()
        if err != nil {
             return nil, err
             klog.Errorf("Error loading incluster config: %+v", err)

      }

       // Loading from config file
       config, err = clientcmd.BuildConfigFromFlags("", "/root/.kube/config")
       if err !=nil{
              klog.Errorf("Error loading config file: %+v", err)
              return nil, fmt.Errorf("Error loading cluster configuration")
        }

```

Reply:
Conversation outdated as the approach changed.
CloudInstanceId is retrieved from the node labels using test suites library instead of using metadata library.